invenio_communities/marshmallow/json.py

```diff
@@ -50,6 +54,32 @@ def serialize_creator(record, context):
     return record.get('created_by', missing)
 
 
+def validate_domain(domain, context):
+    """Load the record creator."""
+    if domain is missing:
+        return missing
+    else:
+        valid_domains = [
+            dom['value'] for dom
+            in LocalProxy(
+                lambda: current_app.config[
+                    'COMMUNITIES_FORM_CONFIG']['domains'])]
```

Comment:
`LocalProxy` shouldn't be necessary here, since `validate_domain` and the marshmallow schema should probably always be called inside an application context.

Reply:
Maybe though, you want to use something like `marshmallow.validate.OneOf`, in which case you should use a `LocalProxy` like so:

```python
from marshmallow import validate, Schema, fields

def domain_choices():
    domains = current_app.config['COMMUNITIES_FORM_CONFIG']['domains']
    return [d['value'] for d in domains]


class Community(Schema):

    domains = SanitizedUnicode(
        validate=validate.OneOf(LocalProxy(domain_choices)),
    )
```